functions-and-scope.js

```diff
@@ -53,23 +86,47 @@ const grades = [9, 8, 5, 7, 7, 4, 9, 8, 8, 3, 6, 8, 5, 6];
 // averageGrade([6, 4, 5]) geeft xxxx
 // averageGrade([8, 9, 4, 6, 10]) geeft xxxx
 
+function averageGrade(grades) {
+    let count = 0
+
+    for (let i = 0; i < grades.length; i++) {
+        count = count + grades[i]
+    }
+
+    const average = count / grades.length
+    return average
+}
+
+const averages = averageGrade(grades)
+console.log(averages);
+
+const averages1 = averageGrade([6, 4, 5])
+console.log(averages1.toFixed(2))
+const averages2 = averageGrade([8, 9, 4, 6, 10])
+console.log(averages2.toFixed(2));
 
 /* 2c: Afronden op twee decimalen */
 // Zorg ervoor dat het gemiddelde cijfer dat wordt teruggegeven uit de functie netjes wordt afgerond op twee decimalen.
 // Tip: Google is your best friend!
 
 
-
-
 /* Bonusopdracht: hoogste cijfer */
 
 /* 3a: Script schrijven  */
 // Schrijf een script die op basis van de grades array (hierboven) checkt wat het hoogst behaalde cijfer is. Je mag hier geen bestaande methoden voor gebruiken. Schrijf de stapjes eerst uit en vraag jezelf de volgende dingen af:
-// * Hoe kan ik iedere waarde van de array langsgaan?
-// * Op welke conditie moet ik checken?
-// * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan opslaan?
+// * Hoe kan ik iedere waarde van de array langsgaan? for loop
+// * Op welke conditie moet ik checken? of i hoger is dan i
+// * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan opslaan? const
 // Log het antwoord in de terminal.
 
+let highestNumber = 0;
```

Comment:
Je kan hier beter niet 0 maar de eerste waarde uit de array pakken. Het zal hier altijd goed gaan, omdat in de praktijk cijfers niet lager dan 0 kunnen zijn. Maar stel je voor dat je een array aanlevert van [-1, -2, -5, -3], dan krijg je 0 terug wat dus niet klopt.